roofit/roofitcore/src/RooAbsReal.cxx

```diff
@@ -2277,25 +2283,42 @@ RooPlot* RooAbsReal::plotOn(RooPlot *frame, PlotOpt o) const
       // Append any suffixes imported from RooAbsPdf::plotOn
       curveName.Append(o.curveNameSuffix) ;
     }
-    curve->SetName(curveName.Data()) ;
+    
+    TString opt(o.drawOptions);
+    if(opt.Contains("P")){
+      RooHist *graph= new RooHist(*projection,*plotVar,1.,1.,frame->getNormVars(),o.errorFR);
```

Comment:
Isn't it advisable to disable evaluation errors similar to what's happening below? Or am I misunderstanding, and the PDF is not being evaluated? Or is it clear that there won't be such errors?

Reply:
you're right, this should be done.